### PR TITLE
Added a warning for when replicated abilities are detected in schema generation

### PR DESCRIPTION
**Pull request by joshuahuburn**
**Contributions**: We are not currently taking public contributions - see our [contributions](CONTRIBUTING.md) policy. However, we are accepting issues and we do want your [feedback](../README.md#give-us-feedback).

-------

#### Description
Added an error message for when we detect replicated gameplay abilities during schema generation. These are not supported yet in the GDK and will cause undefined behaviour.

#### Release note
Feature: Added an error when unsupported replicated gameplay abilities are found in schema generation.

#### Tests
Changed a couple of TestSuite's GameplayAbilities to replicated and was given the following messages after generation:
image

Make a replicated gameplay ability and mark it as replicated. Try generating schema and look for the message.

#### Primary reviewers
@m-samiec @mattyoung-improbable

### CODE REVIEW

File: SpatialGDK/Source/SpatialGDKEditor/SpatialGDKEditor.Build.cs [hunk 1]
```diff
@@ -16,8 +16,9 @@ public SpatialGDKEditor(ReadOnlyTargetRules Target) : base(Target)
 				"Engine",
  				"EngineSettings",
 				"SpatialGDK",
-				"UnrealEd"
-			});
+				"UnrealEd",
+                "GameplayAbilities"
```
Comment (m-samiec) [inline SpatialGDK/Source/SpatialGDKEditor/SpatialGDKEditor.Build.cs]: Tabbing/issue

File: SpatialGDK/Source/SpatialGDKEditor/SpatialGDKEditor.Build.cs [hunk 2]
```diff
@@ -16,8 +16,9 @@ public SpatialGDKEditor(ReadOnlyTargetRules Target) : base(Target)
 				"Engine",
  				"EngineSettings",
 				"SpatialGDK",
-				"UnrealEd"
-			});
+				"UnrealEd",
+				"GameplayAbilities"
+            });
```
Comment (m-samiec) [inline SpatialGDK/Source/SpatialGDKEditor/SpatialGDKEditor.Build.cs]: nit: this line has switched from tabs to spaces. Please revert.

File: SpatialGDK/Source/SpatialGDKEditor/Private/SchemaGenerator/SpatialGDKEditorSchemaGenerator.cpp [hunk 3]
```diff
@@ -393,6 +394,18 @@ TArray<UClass*> GetAllSupportedClasses()
 			}
 		}
 
+		// Check for replicated GameplayAbilities and print a warning if we find one. The UnrealGDK does not currently support this.
+		if (ClassIt->IsChildOf(UGameplayAbility::StaticClass()))
+		{
+			UClass* AbilityClass = *ClassIt;
+			UGameplayAbility* GameplayAbility = Cast<UGameplayAbility>(AbilityClass->GetDefaultObject());
+
+			if (GameplayAbility->GetReplicationPolicy() == EGameplayAbilityReplicationPolicy::ReplicateYes)
```
Comment (improbable-valentyn) [inline SpatialGDK/Source/SpatialGDKEditor/Private/SchemaGenerator/SpatialGDKEditorSchemaGenerator.cpp]: Is it common to change replication policy to `ReplicateYes` at runtime?
If so, we could additionally warn for any ability that has added replicated properties.

Comment (m-samiec) [reply SpatialGDK/Source/SpatialGDKEditor/Private/SchemaGenerator/SpatialGDKEditorSchemaGenerator.cpp]: No shouldn't be. They're instanced from the CDO directly, and used immediately. Modifying the replication state after activation wouldn't function correctly.

**Pull request merged**